lib/IRremoteESP8266/IRremoteESP8266.cpp

```diff
@@ -1314,8 +1303,8 @@ void ICACHE_FLASH_ATTR IRsend::sendMitsubishiAC(unsigned char data[]) {
 // IRRecv------------------------------------------------------
 
 extern "C" {
-  #include "user_interface.h"  // NOLINT(build/include)
   #include "gpio.h"
+  #include "user_interface.h"
```

Comment:
@markszabo Fixed/Done.

Well, that was interesting. Changing them caused more linter issues. When followed, that cause compiler errors, but then following the next (2nd generation) load of linter issues, that solved the compiler errors.

In short, the #include order needs to be:
YourOwnFile.h
"c" system includes
"c++" system includes.
Your other includes.